scripts/test-helm.sh

```diff
@@ -1,95 +1,158 @@
+helm uninstall --namespace "$K8S_NAMESPACE" "$HELM_CHART_NAME" > /dev/null 2>&1
+# cleanup to start the test from scratch
+kubectl delete namespace "$K8S_NAMESPACE" > /dev/null 2>&1
+kubectl delete cluster-role-binding  "ack-$AWS_SERVICE-controller-rolebinding" > /dev/null 2>&1
+kubectl delete cluster-role  "ack-$AWS_SERVICE-controller" > /dev/null 2>&1
+set -e
+pushd "$HELM_DIR" 1> /dev/null
+  # Update values.yaml file
+  yq e "$AWS_REGION_EVAL" -i values.yaml
+  yq e "$AWS_ACCOUNT_ID_EVAL" -i values.yaml
+  yq e "$IMAGE_REPOSITORY_EVAL" -i values.yaml
+  yq e "$IMAGE_TAG_EVAL" -i values.yaml
+
+  echo -n "test-helm.sh] creating k8s namespace $K8S_NAMESPACE ... "
+  kubectl create namespace "$K8S_NAMESPACE" --dry-run=client -o yaml | kubectl apply -f - 1>/dev/null || exit 1
+  echo "ok."
+  echo -n "test-helm.sh] installing the helm chart $HELM_CHART_NAME in namespace $K8S_NAMESPACE ... "
+  helm install --namespace "$K8S_NAMESPACE" "$HELM_CHART_NAME" . 1>/dev/null || exit 1
+  echo "ok."
+  echo -n "test-helm.sh] waiting 10 seconds for $AWS_SERVICE controller to start ... "
+  sleep 10
+  echo "ok"
+  # NOTE: Currently there is only a single pod. Keeping this logic very simple right now.
+  # Update this logic if multiple pods are started in single ack helm chart installation.
+  CONTROLLER_POD_NAME=$(kubectl get pods -n $K8S_NAMESPACE -ojson | jq -r ".items[0].metadata.name")
+  if [ -z "$CONTROLLER_POD_NAME" ]; then
+    echo "test-helm.sh] [ERROR] Found empty ack controller pod name. Exiting ..."
+    exit 1
+  fi
+  echo "test-helm.sh] ack $AWS_SERVICE controller pod name is $K8S_NAMESPACE/$CONTROLLER_POD_NAME"
+  echo -n "test-helm.sh] Verifying that pod status is in Running state ... "
+  POD_STATUS=$(kubectl get pod/"$CONTROLLER_POD_NAME" -n $K8S_NAMESPACE -ojson | jq -r ".status.phase")
+  [[ $POD_STATUS != Running ]] && echo "pod status is $POD_STATUS . Exiting ... " && exit 1
+  echo "ok."
+  echo -n "test-helm.sh] Verifying that there are no ERROR in controller logs ... "
+  CONTROLLER_LOGS=$(kubectl logs pod/"$CONTROLLER_POD_NAME" -n $K8S_NAMESPACE)
+  if [ -z "$CONTROLLER_LOGS" ]; then
+    echo "test-helm.sh] [ERROR] No controller logs found for pod $CONTROLLER_POD_NAME. Exiting ..."
+    exit 1
+  fi
+  if echo "$CONTROLLER_LOGS" | grep -q "ERROR"
+  then
+    echo "test-helm.sh] [ERROR] Found following ERROR statements in controller logs."
+    echo "================================"
+    echo "$CONTROLLER_LOGS" | grep "ERROR"
+    echo "================================"
+    echo "test-helm.sh] [ERROR] Exiting ..."
+    exit 1
+  fi
+  echo "ok."
+  echo -n "test-helm.sh] uninstalling the helm chart $HELM_CHART_NAME in namespace $K8S_NAMESPACE ... "
+  helm uninstall --namespace "$K8S_NAMESPACE" "$HELM_CHART_NAME" 1>/dev/null || exit 1
+  echo "ok."
+  echo -n "test-helm.sh] removing $AWS_SERVICE crds installed by helm ... "
+  kubectl delete -f "$HELM_DIR/crds" 1>/dev/null || exit 1
```

Comment:
No Helm does not.
I added this so that if locally someone runs the release-test and e2e test together, they do not accidentally use CRDs from helm charts in their e2e tests.
Also it's good to cleanup whatever was installed.